src/vs/workbench/api/common/extHostNotebookDocument.ts

```diff
@@ -110,12 +110,30 @@ export class ExtHostCell {
 				output.items.length = 0;
 			}
 			output.items.push(...newItems);
-			if (output.items.every(item => notebookCommon.isTextStreamMime(item.mime))) {
-				const compressed = notebookCommon.compressOutputItemStreams(newOutputItems[0].mime, output.items.map(item => item.data));
+
+			if (output.items.length > 1 && output.items.every(item => notebookCommon.isTextStreamMime(item.mime))) {
+				// Look for the mimes in the items, and keep track of their order.
+				// Merge the streams into one output item, per mime type.
+				const mimeOutputs = new Map<string, Uint8Array[]>();
```

Comment:
Might be useful to share code between here and notebookCellTextModel?

Reply:
@roblourens where would i place such a common (shared) block of code.
I need to share code between the two files
* src/vs/workbench/api/common/extHostNotebookDocument.ts
* src/vs/workbench/contrib/notebook/common/model/notebookCellTextModel.ts

I couldn't find a way to share a file that ends up importing `NotebookCellOutputItem`.
Tried to add such a function in the following locations and got eslint errors about imports not being allowed
* src/vs/workbench/common/notebookCommon.ts
* src/vs/workbench/contrib/notebook/common/notebookCommon.ts

Hgere's the error I get
![Screen Shot 2022-10-12 at 07 12 50](https://user-images.githubusercontent.com/1948812/195189242-473ee35e-a246-4366-b5d7-68a63d09f2d9.png)

Reply:
Yeah, you're already calling into notebookCommon and that makes sense as a place to keep it, but you would have to do it in a way that doesn't directly import 'vscode', eg work with a different interface. Whatever you wanna do though.

Reply:
Oops, missed this, will work on that in debt.